tests/role_wrapper.yml

```diff
@@ -0,0 +1,65 @@
+
+- name: Create a temporary file for a static inventory
+  tempfile:
+    state: file
+    suffix: .inventory.yaml
+  register: tempinventory
+  delegate_to: localhost
+
+- name: Create a static inventory from hostvars
+  template:
+    src: inventory.yaml.j2
+    dest: "{{ tempinventory.path }}"
+  delegate_to: localhost
+
+- name: Import role {{ tests_role_path }}
+  import_role:
+    name: "{{ tests_role_path }}"
+
+- name: >
+    Import role {{ tests_role_path }} via fresh Ansible instance with check
+    mode enabled
+  command: >
+    ansible all -vv -i {{ tempinventory.path }} --check -m import_role
+      -a "name={{ tests_role_path }}"
+  delegate_to: localhost
+  when:
+    - tests_test_checkmode is defined
+    - tests_test_checkmode
+
+- name: Update static inventory with recent hostvars
+  template:
+    src: inventory.yaml.j2
+    dest: "{{ tempinventory.path }}"
+  delegate_to: localhost
+  when:
+    - tests_test_idempotency is defined
+    - tests_test_idempotency
+
+- name: Import role {{ tests_role_path }} via fresh Ansible instance
+  command: >
+    ansible all -vv -i {{ tempinventory.path }} -m import_role 
+      -a "name={{ tests_role_path }}"
+  delegate_to: localhost
+  register: ansible_output
+  when:
+    - tests_test_idempotency is defined
+    - tests_test_idempotency
+
+- name: Check the idempotency
+  assert:
+    that:
+      - ansible_output.stdout.find('| CHANGED => {\n') == -1
+    fail_msg: >
+      Idempotency test failed. Output: {{ ansible_output.stdout_lines }}
+  delegate_to: localhost
+  when:
+    - tests_test_idempotency is defined
+    - tests_test_idempotency
+
+- name: Remove the temporary file
```

Comment:
This should maybe be in a an `always` block to ensure it will still run if something else fails.

Reply:
Fixed in https://github.com/linux-system-roles/timesync/pull/72/commits/9c8124ec7cbca33a2cc98c9bc6f2f842d4ba448b.